Add a .travis.yml file

kienstra opened the pull request:
* Mainly as generated by `wp plugin scaffold <plugin name>`
* Makes some adjustments, like changing the lowest PHP version from `5.3` to `5.4`
* This might need some more changes, like changing the branch from master to something like develop.

kienstra committed:
Add a travis file, mainly as generated with wp plugin scaffold <plugin name>

Makes some adjustments,
like changing the lowest PHP version from 5.3 to 5.4
This might need some more changes,
like changing the branch from master to something like develop.

kienstra committed:
Remove the branch safelist, which looks like only master will be built

According to the Travis documentation:
https://docs.travis-ci.com/user/customizing-the-build/

lukecarbis commented on .travis.yml:
```diff
@@ -0,0 +1,63 @@
+    - $HOME/.composer/cache
+
+matrix:
+  include:
```
We probably only need to test `latest` and `trunk`,

kienstra replied:
OK

kienstra replied:
Sure, https://github.com/getblocklab/block-lab/pull/276/commits/f41885acabb849c45be645209a353c92c104492b removes WordPres 4.5 from the build.

kienstra committed:
Copy the install-wp-tests.sh file from the wp plugin scaffold command

Simply copy and past this
from the result of the command.
It looks like the Travis builds were failing
due to a lack of this.

kienstra committed:
Remove WordPress version 4.5 from the build matrix

This plugin is for the block editor,
which doesn't support WP 4.5.

kienstra committed:
Attempt to address an issue with functions.php not found

Copy 2 blocks from wp-dev-lib,
which might correct the $_tests_dir path.

kienstra committed:
Change the path to the WordPress instance

Instead of /tmp/wordpress-tests,
make it /tmp/wordpress

kienstra committed:
Change the path to wordpress-tests-lib

kienstra committed:
Try removing an extra block that sets $_tests_dir

It might not be needed,
so see if Travis passes without it.
Also, remove the PHPCS command.
That produced an error.
We probably need something like wp-dev-lib
if we're going to run PHPCS,
as it needs to look only at the diff,
not the entire plugin.

kienstra committed:
Move the place where php is passed to phpcs

kienstra committed:
Try moving phpcs.ruleset.xml to phpcs.xml

It looks like that is the default ruleset,
and ideally phpcs would use that in Travis.

kienstra committed:
Move phpcs file to phpcs.xml.dist, change syntax of ignore comments

I had renamed this to phpcs.xml,
but it looks like phpcs recognizes phpcs.xml.dist
as a possible ruleset name.

kienstra commented on phpcs.xml:
```diff
@@ -6,6 +6,10 @@
 
     <config name="testVersion" value="5.4-"/>
 
+    <arg value="s"/>
```
I renamed this so that `phpcs` in Travis will pick this up, without having to pass it as an argument, like `phpcs --standard=phpcs.ruleset.xml`

kienstra replied:
If this name is important, we could keep the name and just pass it as an argument in `.travis.yml`
